spec/services/filtered_artwork_search_service_spec.rb

```diff
@@ -0,0 +1,50 @@
+require 'spec_helper'
+require './app/services/filtered_artwork_search_service'
+
+RSpec.describe FilteredArtworkSearchService, type: :model do
+  describe '.call' do
+    let(:endpoint) { "#{Rails.application.config_for(:gravity)['api_root']}/filter/artworks" }
+
+    let(:params) do
+      # kitchen-sink query that should match kiki-smith-the-falls-i
+      {
+        keyword: 'tattoo',
+        gene_ids: ['mixed-media'],
+        tag_id: 'waterfall',
+        artist_ids: ['kiki-smith'],
+        partner_id: 'universal-limited-art-editions',
+        fair_id: 'art-basel-2017',
+        attribution_class: ['limited edition'],
+        # TODO: creation date # (2013-11-06T19:01:19Z)
+        price_range: '5000-10000',
+        # TODO: ids # (527a91ffcd530e5006000400)
+        # TODO: include_published # (requires user access token),
+        acquireable: true,
+        offerable: true,
+        for_sale: true
+      }
+    end
+
+    let(:artwork_hit) do
+      JSON.parse(File.read('./spec/fixtures/kiki-smith-the-falls-i.json'))
+    end
+
+    let(:success_response) do
+      { hits: [artwork_hit], aggregations: [] }
+    end
+
+    let!(:filter_artworks_url) do
+      WebMock.stub_request(:get, endpoint).with(query: params).to_return(status: 200, body: success_response.to_json, headers: {})
+    end
```

Comment:
**suggestion / non-blocking**: What do you think of defining a helper method instead of putting this in a WebMock setup in a `let`?

```ruby
it 'issues the correct gravity request' do
  request = stub_filter_artworks_request(200, success_response)

  FilteredArtworkSearchService.call(params)

  expect(request).to have_been_requested
end

it 'returns the artwork hits' do
  stub_filter_artworks_request(401, unauthorized_response)

  response = FilteredArtworkSearchService.call(params)

  expect(response['error']).to eq "Unauthorized"
end
```

Reply:
Yep, I like both suggestions 👍🏽 

Re-worked the spec in https://github.com/artsy/rosalind/pull/669/commits/4b19de39fcd917ad12450143c503f62d9199b816